src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocBlockTagLocationCheck.java

```diff
@@ -0,0 +1,199 @@
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocBlockTagLocation"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * &#47;**
+ *  * Escaped tag &amp;#64;version (OK)
+ *  * Plain text with {&#64;code &#64;see} (OK)
+ *  * A @custom tag (OK)
+ *  * &#64;since 1.0 (OK)
+ *  * text &#64;return (violation)
+ *  * * &#64;param (violation)
+ * +* &#64;serial (violation)
+ *  * &#64;see first (OK) &#64;see second (violation)
+ *  *&#47;
+ * public int field;
+ * </pre>
+ * <p>
+ * To configure the check to verify tags from
+ * <a href="https://openjdk.java.net/jeps/8068562">JEP 8068562</a> only:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocBlockTagLocation"&gt;
+ *   &lt;property name="tags" value="apiNote, implSpec, implNote"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>
+ * To configure the check to verify all default tags and some custom tags in addition:
+ * </p>
+ * <pre>
+ * &lt;module name="JavadocBlockTagLocation"&gt;
+ *   &lt;!-- default tags --&gt;
+ *   &lt;property name="tags" value="author, deprecated, exception, hidden"/&gt;
+ *   &lt;property name="tags" value="param, provides, return, see, serial"/&gt;
+ *   &lt;property name="tags" value="serialData, serialField, since, throws"/&gt;
+ *   &lt;property name="tags" value="uses, version"/&gt;
+ *   &lt;!-- additional tags used in the project --&gt;
+ *   &lt;property name="tags" value="noinspection"/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ *
+ * @since 8.24
+ */
+@StatelessCheck
+public class JavadocBlockTagLocationCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties" file.
+     */
+    public static final String MSG_BLOCK_TAG_LOCATION = "javadoc.blockTagLocation";
+
+    /**
+     * This regexp is used to extract the javadoc tags.
+     */
+    private static final Pattern JAVADOC_BLOCK_TAG_PATTERN = Pattern.compile("\\s@(\\w+)");
+
+    /**
+     * Block tags from Java 11
+     * <a href="https://docs.oracle.com/en/java/javase/11/docs/specs/doc-comment-spec.html>
+     * Documentation Comment Specification</a>.
+     */
+    private static final String[] DEFAULT_TAGS = {
+        "author",
+        "deprecated",
+        "exception",
+        "hidden",
+        "param",
+        "provides",
+        "return",
+        "see",
+        "serial",
+        "serialData",
+        "serialField",
+        "since",
+        "throws",
+        "uses",
+        "version",
+    };
+
+    /**
+     * Specify the javadoc tags to process.
+     */
+    private Set<String> tags;
+
+    /**
+     * Creates a new {@code JavadocBlockTagLocationCheck} instance with default settings.
+     */
+    public JavadocBlockTagLocationCheck() {
+        setTags(DEFAULT_TAGS);
+    }
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.TEXT,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
```

Comment:
please make getRequiredJavadocTokens to return list of tokens.
please override getAcceptableJavadocTokens to in body of method use getRequiredJavadocTokens.
getDefaultJavadocTokens should use getRequiredJavadocTokens.

`getRequired.*Tokens` is most minimal set of tokens so it should be used in other method , not in different way.

Please add comment above getRequiredJavadocTokens to explain that we Check is working against specification of javadoc and need to find non-escaped tags from text, so all text should be re-parsed find tags. It is better to share link to spec (the same as you did in github comment)

Reply:
done